pyo3-ffi/src/datetime.rs

```diff
@@ -739,8 +747,13 @@ extern "C" {
 
 // Rust specific implementation details
 
-struct PyDateTimeAPISingleton(UnsafeCell<*mut PyDateTime_CAPI>);
+struct PyDateTimeAPISingleton {
+    once: Once,
+    ptr: UnsafeCell<*mut PyDateTime_CAPI>,
+}
 unsafe impl Sync for PyDateTimeAPISingleton {}
 
-static PyDateTimeAPI_impl: PyDateTimeAPISingleton =
-    PyDateTimeAPISingleton(UnsafeCell::new(ptr::null_mut()));
+static PyDateTimeAPI_impl: PyDateTimeAPISingleton = PyDateTimeAPISingleton {
+    once: Once::new(),
+    ptr: UnsafeCell::new(ptr::null_mut()),
```

Comment:
There might be an alternative implementation which uses an `AtomicPtr<PyDateTime_CAPI>` instead of a once, with a `compare_exchange` call to write a nonzero value? I think there's unlikely to be any practical difference given that the once (I _think_) stores its initialization state as an atomic value.

Reply:
Makes sense. Let's keep the `Once` 👍